lib/services/livesync/livesync-service-base.ts

```diff
@@ -176,53 +175,45 @@ class LiveSyncServiceBase implements ILiveSyncServiceBase {
 			return (() => {
 				let shouldRefreshApplication = true;
 				let deviceAppData = this.$deviceAppDataFactory.create(appIdentifier, this.$mobileHelper.normalizePlatformName(platform), device, liveSyncOptions);
-				if (deviceAppData.isLiveSyncSupported().wait()) {
-					let platformLiveSyncService = this.resolvePlatformLiveSyncService(platform, device);
+				let platformLiveSyncService = this.resolvePlatformLiveSyncService(platform, device);
 
-					if (platformLiveSyncService.beforeLiveSyncAction) {
-						platformLiveSyncService.beforeLiveSyncAction(deviceAppData).wait();
-					}
+				if (platformLiveSyncService.beforeLiveSyncAction) {
+					platformLiveSyncService.beforeLiveSyncAction(deviceAppData).wait();
+				}
 
-					// Not installed application
-					device.applicationManager.checkForApplicationUpdates().wait();
+				// Not installed application
+				device.applicationManager.checkForApplicationUpdates().wait();
 
-					let wasInstalled = true;
-					if (!device.applicationManager.isApplicationInstalled(appIdentifier).wait() && !this.$options.companion) {
-						this.$logger.warn(`The application with id "${appIdentifier}" is not installed on device with identifier ${device.deviceInfo.identifier}.`);
-						if (!packageFilePath) {
-							packageFilePath = this.$liveSyncProvider.buildForDevice(device).wait();
-						}
-						device.applicationManager.installApplication(packageFilePath).wait();
+				let wasInstalled = true;
+				if (!device.applicationManager.isApplicationInstalled(appIdentifier).wait()) {
+					this.$logger.warn(`The application with id "${appIdentifier}" is not installed on device with identifier ${device.deviceInfo.identifier}.`);
+					if (!packageFilePath) {
+						packageFilePath = this.$liveSyncProvider.buildForDevice(device).wait();
+					}
+					device.applicationManager.installApplication(packageFilePath).wait();
 
-						if (platformLiveSyncService.afterInstallApplicationAction) {
```

Comment:
afterInstallApplicationAction is implemented in both platforms so this condition should always be true. So I've removed the else part. ping @tzraikov if I am missing something